ingress/controllers/nginx-third-party/controller.go

```diff
@@ -291,58 +287,24 @@ func (lbc *loadBalancerController) getUpstreamServers(data []interface{}) ([]ngi
 					}
 				}
 
-				upstreams[name] = ups
-			}
-		}
-
-		pems := lbc.getPemsFromIngress(data)
-
-		for _, rule := range ing.Spec.Rules {
-			var server nginx.Server
-			if existent, ok := servers[rule.Host]; ok {
-				server = existent
-			} else {
-				server = nginx.Server{Name: rule.Host}
-			}
-
-			if pemFile, ok := pems[rule.Host]; ok {
-				server.SSL = true
-				server.SSLCertificate = pemFile
-				server.SSLCertificateKey = pemFile
-			}
-
-			var locations []nginx.Location
-
-			for _, path := range rule.HTTP.Paths {
-				loc := nginx.Location{Path: path.Path}
-				upsName := ing.GetNamespace() + "-" + path.Backend.ServiceName
-
-				svcKey := ing.GetNamespace() + "/" + path.Backend.ServiceName
-				_, svcExists, err := lbc.svcLister.Store.GetByKey(svcKey)
-				if err != nil {
-					glog.Infof("error getting service %v from the cache: %v", svcKey, err)
-					continue
-				}
-
-				if !svcExists {
-					glog.Warningf("service %v does no exists. skipping Ingress rule", svcKey)
-					continue
-				}
-
 				for _, ups := range upstreams {
 					if upsName == ups.Name {
-						loc.Upstream = ups
+						loc := nginx.Location{Path: path.Path}
+						loc.Upstream = *ups
+						locations = append(locations, loc)
+						break
 					}
 				}
-				locations = append(locations, loc)
 			}
 
 			server.Locations = append(server.Locations, locations...)
-			servers[rule.Host] = server
 		}
 	}
 
-	aUpstreams := make([]nginx.Upstream, 0, len(upstreams))
+	// TODO: find a way to make this more readable
+	// The structs must be ordered to always generate the same file
+	// if the content does not change.
+	aUpstreams := make([]*nginx.Upstream, 0, len(upstreams))
```

Comment:
please add a unittest for this if you don't already have one (i mean different order of services and endpoints produce the same config file hash).